src/c_interface.rs

```diff
@@ -462,6 +462,30 @@ macro_rules! c_accessor_string {
     };
 }
 
+macro_rules! c_accessor_string_mut {
+    ($(#[$($attrss:tt)*])* $rust:ident, $rust_set:ident, $c:ident) => {
+        $(#[$($attrss)*])*
+        #[must_use]
+        pub fn $rust(&self) -> &str {
+            unsafe {
+                if !self.c_ptr_ref().$c.is_null() {
+                    crate::c_interface::from_c_str(std::ffi::CStr::from_ptr(self.c_ptr_ref().$c))
+                } else {
+                    ""
+                }
+            }
+        }
+
+        $(#[$($attrss)*])*
+        pub fn $rust_set(&mut self, value: String) {
+            let c_str = std::ffi::CString::new(value).expect("Null byte found in provided string");
```

Comment:
I realize it's an exceedingly rare case, but I think we should bubble up the error here, wrapped in `SpineError::NulError`.